src/app/hooks.ts

```diff
@@ -0,0 +1,5 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from './store';
+
+export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
```

Comment:
The `useDispatch.withTypes<AppDispatch>()` syntax is incorrect. You should use `useDispatch<AppDispatch>()` to create a typed version of the dispatch hook.